index.js

```diff
@@ -186,6 +188,12 @@ HomeAssistantPlatform.prototype = {
           accessory = new HomeAssistantSwitch(that.log, entity, that, 'input_boolean')
         }else if (entity_type == 'fan'){
           accessory = new HomeAssistantFan(that.log, entity, that)
+        }else if (entity_type == 'binary_sensor'){
+          if (entity.attributes && entity.attributes.sensor_class == 'opening') {
+            accessory = new HomeAssistantBinarySensor(that.log, entity, that, Service.Door, Characteristic.CurrentPosition, 100, 0)
```

Comment:
Throwing out some other super easy to implement binary_sensors for extra credit:

- `moisture` - [`Service.LeakSensor`](https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L2413-L2429)
- `occupancy ` - [`Service.OccupancySensor`](https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L2560-L2580)
- `smoke ` - [`Service.SmokeSensor`](https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L2679-L2699)